emhttp/plugins/extract_translations.php

```
@@ -0,0 +1,285 @@
+            $beforeMatch = substr($content, 0, $offset);
+            $lineNumber = substr_count($beforeMatch, "\n") + 1;
+            
+            // Remove surrounding quotes if present
+            if ((substr($text, 0, 1) === '"' && substr($text, -1) === '"') ||
+                (substr($text, 0, 1) === "'" && substr($text, -1) === "'")) {
+                $text = substr($text, 1, -1);
+            }
+            
+            // Filter out obvious regex patterns and false positives
+            // Skip if it's mostly regex special characters or looks like a regex pattern
+            $specialChars = preg_match_all('/[\/\^\$\*\+\?\{\}\[\]\\\|]/', $text, $temp);
```

PHPMD (2.15.0)

Avoid unused local variables such as '$temp'. (undefined)

(UnusedLocalVariable)